pandas/_libs/lib.pyx

```diff
@@ -2270,7 +2270,7 @@ def map_infer_mask(ndarray arr, object f, const uint8_t[:] mask, bint convert=1,
     result = np.empty(n, dtype=dtype)
     for i in range(n):
         if mask[i]:
-            if na_value is _no_default:
+            if na_value is no_default:
```

Comment:
having a cdef version (kind of like we have for NaT) would help perf here (not necessary for this PR)